### PR TITLE
Address outstanding issues around custom modifiers in VB.

### LINKED ISSUE #8948: Investigate and implement correct propagation of modopts in byref members
C# does not define modopts, but we could be overriding/implementing an existing byref returning method with modopts. (  long& Foo()  in managed c++ ? )

We need to ensure that the modifiers in the signature are correctly propagated to the overriding/implementing method.

### LINKED ISSUE #14453: ReturnType CustomModifiers are simply copied from overridden method
This is not correct when overridden method is generic. The modifiers might refer to its type parameters, they should be substituted with corresponding type parameters of the overriding method.

Should check similar scenario with interface implementation.

### PR DESCRIPTION
**Pull request by AlekseyTs**
This change fixes the following issues in VB compiler: #8948 and #14453.
Closes #8948. Closes #14453.

@dotnet/roslyn-compiler, @VSadov, @cston  Please review.

### CODE REVIEW

**Comment by genlu** [comment]
@dotnet-bot retest windows_release_unit32_prtest please

**Comment by AlekseyTs** [comment]
@dotnet/roslyn-compiler, @VSadov, @cston Please review.

File: src/Compilers/VisualBasic/Portable/SymbolDisplay/SymbolDisplayVisitor.Members.vb [hunk 1]
```diff
@@ -97,13 +97,29 @@ Namespace Microsoft.CodeAnalysis.VisualBasic
...
 
+            Dim vbProperty = TryCast(symbol, PropertySymbol)
+            Dim countOfCustomModifiersPrecedingByRef As UShort = If(vbProperty Is Nothing, 0US, vbProperty.CountOfCustomModifiersPrecedingByRef)
+
+            If symbol.ReturnsByRef AndAlso format.MemberOptions.IncludesOption(SymbolDisplayMemberOptions.IncludeRef) Then
+                AddSpace()
+                AddKeyword(SyntaxKind.ByRefKeyword)
+
+                If countOfCustomModifiersPrecedingByRef > 0 Then
+                    AddCustomModifiersIfRequired(symbol.TypeCustomModifiers.Take(countOfCustomModifiersPrecedingByRef).AsImmutable())
```
Comment (cston) [inline src/Compilers/VisualBasic/Portable/SymbolDisplay/SymbolDisplayVisitor.Members.vb]: Perhaps `AddCustomModifiersIfRequired(ImmutableArray.Create(symbol.TypeCustomModifiers, 0, countOfCustomModifiersPrecedingByRef))`. Same comment below.

**Comment by cston** [comment]
LGTM

File: src/Compilers/VisualBasic/Portable/SymbolDisplay/SymbolDisplayVisitor.Members.vb [hunk 2]
```diff
@@ -97,13 +97,29 @@ Namespace Microsoft.CodeAnalysis.VisualBasic
                 End If
             End If
 
+            Dim vbProperty = TryCast(symbol, PropertySymbol)
+            Dim countOfCustomModifiersPrecedingByRef As UShort = If(vbProperty Is Nothing, 0US, vbProperty.CountOfCustomModifiersPrecedingByRef)
```
Comment (AlekseyTs) [inline src/Compilers/VisualBasic/Portable/SymbolDisplay/SymbolDisplayVisitor.Members.vb]: We have just decided to expose this information through the public API. I will remove this code in the process in a separate change.

File: src/Compilers/VisualBasic/Portable/SymbolDisplay/SymbolDisplayVisitor.Members.vb (hunk 1 above)
Comment (AlekseyTs) [inline src/Compilers/VisualBasic/Portable/SymbolDisplay/SymbolDisplayVisitor.Members.vb]: This code will be changed along with an introduction of a new public API, in a separate change.

File: src/Compilers/VisualBasic/Test/Emit/CodeGen/CodeGenRefReturnTests.vb [hunk 3]
```diff
@@ -88,7 +88,7 @@ End Module",
   IL_0012:  stloc.0
   IL_0013:  ldloc.0
   IL_0014:  ldc.i4.2
-  IL_0015:  call       "Function C.F(Of Integer)(Integer(), Integer) As Integer"
+  IL_0015:  call       "Function C.F(Of Integer)(Integer(), Integer) ByRef As Integer"
```
Comment (VSadov) [inline src/Compilers/VisualBasic/Test/Emit/CodeGen/CodeGenRefReturnTests.vb]: I think this should be "ByRef Function C.F . . .  "

**Comment by VSadov** [comment]
LGTM except for the ByRef  method/property/delegate visualization

File: src/Compilers/VisualBasic/Test/Emit/CodeGen/CodeGenRefReturnTests.vb (hunk 3 above)
Comment (AlekseyTs) [inline src/Compilers/VisualBasic/Test/Emit/CodeGen/CodeGenRefReturnTests.vb]: Yes, I think we settled on this form of display. I am going to address this in a separate change.

**Pull request merged**